### PR TITLE
implement test for constructIngressFromCluster

### PR DESCRIPTION
**Pull request by Treenhan**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
We need this test to get full coverage on all methods that get ingresses from different sources (e.g. from file and from cluster)

Fixes https://github.com/kubernetes-sigs/ingress2gateway/issues/48

```release-note
NONE
```

### CODE REVIEW

**Comment by levikobi** [comment]
Thanks @Treenhan 

/lgtm
/assign @LiorLieberman

File: pkg/i2gw/ingress2gateway_test.go [hunk 1]
```diff
@@ -158,3 +160,51 @@ func compareIngressLists(t *testing.T, gotIngressList *networkingv1.IngressList,
...
+	}, {
+		name:        "Test cluster client without resources",
+		runtimeObjs: []runtime.Object{},
+		wantIngress: []networkingv1.Ingress{},
+	},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ingressList := &networkingv1.IngressList{}
```
Comment (LiorLieberman) [inline pkg/i2gw/ingress2gateway_test.go:196]: why not calling this `got` initially?

File: pkg/i2gw/ingress2gateway_test.go [hunk 2]
```diff
@@ -158,3 +160,51 @@ func compareIngressLists(t *testing.T, gotIngressList *networkingv1.IngressList,
...
+		runtimeObjs []runtime.Object
+		wantIngress []networkingv1.Ingress
+	}{{
+		name:        "Test cluster client with 2 resources",
+		runtimeObjs: []runtime.Object{&ingress1, &ingress2},
+		wantIngress: []networkingv1.Ingress{ingress1, ingress2},
+	}, {
+		name:        "Test cluster client without resources",
+		runtimeObjs: []runtime.Object{},
+		wantIngress: []networkingv1.Ingress{},
```
Comment (LiorLieberman) [inline pkg/i2gw/ingress2gateway_test.go:190]: should be plural

File: pkg/i2gw/ingress2gateway_test.go [hunk 3]
```diff
@@ -158,3 +160,51 @@ func compareIngressLists(t *testing.T, gotIngressList *networkingv1.IngressList,
...
+			ingressList := &networkingv1.IngressList{}
+			cl := fake.NewClientBuilder().WithRuntimeObjects(tc.runtimeObjs...).Build()
+			err := ConstructIngressesFromCluster(cl, ingressList)
+			if err != nil {
+				t.Errorf("test failed unexpectedly: %v", err)
+			}
+
+			got := []networkingv1.Ingress{}
+			got = append(got, ingressList.Items...)
+			if !apiequality.Semantic.DeepEqual(tc.wantIngress, got) {
```
Comment (LiorLieberman) [inline pkg/i2gw/ingress2gateway_test.go:205]: maybe apply ingresses1, 2 and and noNamespace that you already have above and use `compareIngressLists` function?

**Comment by LiorLieberman** [comment]
Thanks @Treenhan !
/lgtm
/approve

**Pull request merged**